napari/utils/transforms/transforms.py

```diff
@@ -106,8 +113,10 @@ def _is_diagonal(self):
         return False
 
     def _clean_cache(self):
+        self._cache_dict.clear()
         cached_properties = ('_is_diagonal',)
         [self.__dict__.pop(p, None) for p in cached_properties]
```

Comment:
We're currently using `cached_property` for `_is_diagonal` and clearing it appropriately here.

Using `_cache_dict` introduces a different caching mechanism.

I don't have a strong preference for which mechanism we use, but we should only use one inside this module.

Reply:
done